pkg/util/proxy_test.go

```diff
@@ -172,15 +172,16 @@ func Test_parse_special_character_uri(t *testing.T) {
 	var urlList = []struct {
 		givenURI    string
 		expectedURI string
+		scheme      string
 	}{
-		{"", ""},
-		{"http://foo.com:3128", "http://foo.com:3128"},
-		{"http://user:F@oo!B#ar$@myserver:3128", "http://user:F@oo%21B%23ar$@myserver:3128"},
-		{"https://myuser:my#pass@foo.com:3128", "https://myuser:my%23pass@foo.com:3128"},
-		{"https://newuser:new(pas*)wrd@test.com:3128", "https://newuser:new%28pas%2A%29wrd@test.com:3128"},
+		{"", "", "http"},
+		{"http://foo.com:3128", "http://foo.com:3128", "http"},
+		{"http://user:F@oo!B#ar$@myserver:3128", "http://user:F@oo%21B%23ar$@myserver:3128", "http"},
+		{"https://myuser:my#pass@foo.com:3128", "https://myuser:my%23pass@foo.com:3128", "https"},
+		{"https://newuser:new(pas*)wrd@test.com:3128", "https://newuser:new%28pas%2A%29wrd@test.com:3128", "https"},
```

Comment:
@gbraad shouldn't we test for  something like`{"foo.com:3128", "http://foo.com:3128", "http"}` ?

Reply:
Added, but this problem does not occur here. Let's see what CI says...